app/components/post/post_message.js

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Platform,
+    StyleSheet,
+    TouchableOpacity,
+    View
+} from 'react-native';
+import Icon from 'react-native-vector-icons/Ionicons';
+
+import FileAttachmentList from 'app/components/file_attachment_list';
+import FormattedText from 'app/components/formatted_text';
+import Markdown from 'app/components/markdown';
+import OptionsContext from 'app/components/options_context';
+import SlackAttachments from 'app/components/slack_attachments';
+import {getMarkdownTextStyles, getMarkdownBlockStyles} from 'app/utils/markdown';
+import {makeStyleSheetFromTheme} from 'app/utils/theme';
+
+export default class PostMessage extends PureComponent {
+    static propTypes = {
+        attachments: PropTypes.array,
+        canDelete: PropTypes.bool,
+        canEdit: PropTypes.bool,
+        fileIds: PropTypes.array,
+        hasBeenDeleted: PropTypes.bool,
+        intl: PropTypes.object.isRequired,
+        isFlagged: PropTypes.bool,
+        isPendingOrFailedPost: PropTypes.bool,
+        isSystemMessage: PropTypes.bool,
+        flagPost: PropTypes.func,
+        message: PropTypes.string,
+        navigator: PropTypes.object.isRequired,
+        onFailedPostPress: PropTypes.func,
+        onPostDelete: PropTypes.func,
+        onPostEdit: PropTypes.func,
+        onPress: PropTypes.func,
+        postId: PropTypes.string.isRequired,
+        renderReplyBar: PropTypes.func,
+        showReplyBar: PropTypes.bool,
+        theme: PropTypes.object,
+        toggleSelected: PropTypes.func,
+        unflagPost: PropTypes.func
+    };
+
+    static defaultProps = {
+        fileIds: [],
+        flagPost: () => true,
+        onFailedPostPress: () => true,
+        onPostDelete: () => true,
+        onPostEdit: () => true,
+        onPress: () => true,
+        renderReplyBar: () => true,
+        toggleSelected: () => true,
+        unflagPost: () => true
+    };
+
+    hideOptionsContext = () => {
+        if (Platform.OS === 'ios') {
+            this.refs.options.hide();
+        }
+    };
+
+    showOptionsContext = () => {
+        return this.refs.options.show();
```

Comment:
Instead of using refs, could you just make the `OptionsContext` do this itself? Right now, you're passing it a function that just calls one of its own methods